[Style guide] Grammar and Formatting (#9263)
cloudflare/cloudflare-docs: content/style-guide/grammar-and-formatting/code-conventions-and-format.md

---
pcx_content_type: Concept
title: Code Conventions and Format
---

## Code Conventions and Format

Use the conventions described below throughout Cloudflare product content.

## Angle brackets ( < and > )

Use angle brackets as a placeholder for variables you want the user to enter (except in URLs, where you should use curly braces for placeholders).

For example:

Type your user-specified domain in this format: `https://<user-specified domain>.cloudflare.com`

```
{
  "description": "<RULE_DESCRIPTION>"
}
```

Angle brackets that contain numbers separated by an ellipsis represent a range of values associated with a bit or single name - for example, AO <0...3>.

## Square brackets ( [ and ] )

Square brackets enclose optional items.

For example:

Specify a subsearch that starts with this search command: ``` tag=dns query [search tag=malware]. ```

## Curly braces ( { and } )

Use curly braces in the following situations:

When they are part of a code sample or other string literal.
In placeholders inside a URL: ``` https://api.cloudflare.com/client/v4/organizations/{organization_identifier}/invites ```

## >

> The > symbol leads you through nested menu items and dialog box options to a final action. The sequence **Options > Settings > General** directs you to pull down the **Options** menu, select the **Settings** item, and select **General** from the last dialog box. Do not use bold formatting for the > symbol (refer to "bold" in this table).

Reviewer: ```suggestion
The > symbol leads you through nested menu items and dialog box options to a final action. The sequence **Options > Settings > General** directs you to pull down the **Options** menu, select the **Settings** item, and select **General** from the last dialog box. Do not use bold formatting for the > symbol.
```

## Tip icon

This icon denotes a tip, which alerts you to advisory information.

## Note icon

This icon denotes a note, which alerts you to important information.

## Info icon

This icon denotes info, which alerts you to important information.

## Notice icon

This icon denotes a notice, which alerts you to take precautions to avoid data loss, loss of signal integrity, or degradation of performance.

## Caution icon

> This icon denotes a caution, which advise you to take precautions to avoid injury.

Reviewer: ```suggestion
This icon denotes a caution, which advises you to take precautions to avoid injury.
```

## Blue text

Text in this color indicates a link.

## **bold**

> Use **bold** when referring to a clickable action or to highlight a title/name/etc in the UI. Bold text denotes items that you must select or click in the software, such as menu items, toggles, and dialog box options. Bold text denotes titles or names that are identifiers in the UI. Bold text also denotes a parameter name. 

Reviewer: ```suggestion
Use **bold** when referring to a clickable action or to highlight a title or name in the UI. Bold text denotes items that you must select or click in the software, such as menu items, toggles, and dialog box options. Bold text denotes titles or names that are identifiers in the UI. Bold text also denotes a parameter name. 
```

Reviewer: I feel like this paragraph is a bit repetitive, can you condense it/combine it together?

> Do not use for programs.

Reviewer: ```suggestion
Do not use **bold** for programs.
```

> In nested menus, use bold for the word not the symbol. Example: **Dashboard** > **This** > **That**

Reviewer: We should remain consistent with example lines.
I noticed some pages use "Example:" while others use "For example,"
Wdyt?

## *italics*

Use *italics* when referring to an option that customers can select from, like in dropdown menus.

Do not use italics when referring to the state of a toggle - for example, enabled/disabled should not be italicized. 

## `monospace`

``` `text in between backticks` ```

> Text in this font denotes text or characters that you should enter from the keyboard, sections of code, programming examples, and syntax examples. This font is also used for the proper names of drives, paths, directories, programs, subprograms, devices, functions, operations, variables, files, API commands, and extensions. Examples of elements we monospace:

Reviewer: ```suggestion
Text in this font denotes text or characters that you should enter from the keyboard, sections of code, programming examples, and syntax examples. This font is also used for the proper names of drives, paths, directories, programs, subprograms, devices, functions, operations, variables, files, API commands, and extensions. 

Examples of elements we monospace:
```

Reviewer: Perhaps we can make this an H3

---
pcx_content_type: Concept
title: Code conventions and format
---

# Code conventions and format

Use the conventions described below throughout Cloudflare product content.

## Angle brackets ( < and > )

Use angle brackets as a placeholder for variables you want the user to enter (except in URLs, where you should use curly braces for placeholders).

Example:

Type your user-specified domain in this format: `https://<user-specified domain>.cloudflare.com`

```
{
  "description": "<RULE_DESCRIPTION>"
}
```

Angle brackets that contain numbers separated by an ellipsis represent a range of values associated with a bit or single name - for example, AO <0...3>.

## Square brackets ( [ and ] )

Square brackets enclose optional items.

Example:

> Specify a subsearch that starts with this search command: ``` tag=dns query [search tag=malware]. ```

Reviewer: I don't think you'd need more than one pair of backticks when monospacing one-line phrases.
```suggestion
Specify a subsearch that starts with this search command: `tag=dns query [search tag=malware].`
```

## Curly braces ( { and } )

Use curly braces in the following situations:

> When they are part of a code sample or other string literal.
> In placeholders inside a URL: ``` https://api.cloudflare.com/client/v4/organizations/{organization_identifier}/invites ```

Reviewer: ```suggestion
When they are part of a code sample or other string literal, such as in placeholders inside a URL.

Example:

``` https://api.cloudflare.com/client/v4/organizations/{organization_identifier}/invites ```
```

## >

The > symbol leads you through nested menu items and dialog box options to a final action. The sequence **Options > Settings > General** directs you to pull down the **Options** menu, select the **Settings** item, and select **General** from the last dialog box. Do not use bold formatting for the > symbol.

## Tip icon

This icon denotes a tip, which alerts you to advisory information.

## Note icon

This icon denotes a note, which alerts you to important information.

## Info icon

This icon denotes info, which alerts you to important information.

## Notice icon

This icon denotes a notice, which alerts you to take precautions to avoid data loss, loss of signal integrity, or degradation of performance.

## Caution icon

This icon denotes a caution, which advises you to take precautions to avoid injury.

## Blue text

Text in this color indicates a link.

## **Bold**

Use **bold** when referring to a clickable action or to highlight a title or name in the UI. Bold text denotes items that you must select or click in the software, identifiers in the UI, or parameter names.

> Do not bold use for programs.

Reviewer: ```suggestion
Do not use bold for programs.
```

In nested menus, use bold for the word not the symbol.

Example: **Dashboard** > **This** > **That**

## *Italics*

Use *italics* when referring to an option that customers can select from, like in dropdown menus.

Do not use italics when referring to the state of a toggle - for example, enabled/disabled should not be italicized. 

## `Monospace`

``` `text in between backticks` ```

Text in this font denotes text or characters that you should enter from the keyboard, sections of code, programming examples, and syntax examples. This font is also used for the proper names of drives, paths, directories, programs, subprograms, devices, functions, operations, variables, files, API commands, and extensions. 

> ## Examples of elements we monospace:

Reviewer: ```suggestion
### Examples of elements we monospace
```

---
pcx_content_type: Concept
title: Code Conventions and Format
---

## Code Conventions and Format

> Use the conventions described in this topic throughout Cloudflare product content.

Reviewer: ```suggestion
Use the conventions described below throughout Cloudflare product content.
```

## Angle brackets ( < and > )

Use angle brackets as a placeholder for variables you want the user to enter (except in URLs, where you should use curly braces for placeholders).

> 

Reviewer: ```suggestion
```
